### DISCUSSION TITLE
Multiple APIs support

### DISCUSSION THREAD

**Discussion by Andlix** [universam1/iSpindel#539] [Ideas]
I think it doesn't consume much battery if the iSpindel calls two APIs.
But I could e.g. connect the iSpindel with Brewfather and my Raspberry.

**Comment by pppedrillo** [comment]
No need to abuse iSpindel - it makes its configuration process complicated (ie each service would have to be configured separately at the same time)
Instead you can turn your Rasp into sort of proxy - eg make iSpindel to upload its data to Raspbery and, in turn, the Raspberry to upload the data to BF.

**Reply by b1tman1ac** [reply]
This would be a useful feature. Tilt has this. Some examples would be sending to Brewfather and controlling a temperature controller at the same time.

If this isn't possible providing documentation on an API proxy would be useful that has been tested and functional with all ispindel endpoints.

**Reply by thegreatgunbantoad** [reply]
Certainly doable, but not a trivial update.  I'd be inclined to add a multiple a API option, at which point you'd serve a different config page that can then drop into a setup page for each API type (Maybe just one page for them all).  For most users they wouldn't notice much had changed.  If I get a bit of free time over Christmas I might look at how much work it is likely to be, maybe even have a bash at it.

**Reply by thegreatgunbantoad** [reply]
Had a bit of a look, it's a hell of a lot of work.  Would have to do some major work to get it to store into different data structs for each API type.  Brewfather allows you to forward the data on to another API I think.

**Reply by Hwerow** [reply]
It would be good to be able to select to send to 2 different places say Brewfather and MQTT. This would allow both remote and local monitoring/control feedback.
